code/datums/components/ai_storage.dm

```diff
@@ -0,0 +1,28 @@
+/datum/component/ai_storage
+	var/mob/living/silicon/ai/AI
+	var/wiping = FALSE
+
+/datum/component/ai_storage/Initialize(_wiping=FALSE)
+	wiping = _wiping
+	RegisterSignal(COMSIG_ITEM_ATTACK_OBJ, .proc/attack_obj)
+
+/datum/component/ai_storage/proc/wipe()
+	INVOKE_ASYNC(src, .proc/wipe_loop)

```

Comment:
There's no reason not to just have the contents of `wipe_loop()` in here, isn't there? This is literally a proc whose purpose is to call another proc!

Reply:
It's purpose is to INVOKE_ASYNC the proc, as INVOKE_ASYNC allows the current proc to continue.

Reply:
#33446, change it